reporter/symbol_uploader.go

```diff
@@ -487,6 +534,56 @@ func HasDWARFData(f *pfelf.File) bool {
 	return len(f.Progs) == 0 && hasBuildID && hasDebugStr
 }
 
+func findGoPCLnTab(ef *pfelf.File) (goPCLnTabData, error) {
+	var err error
+	var data []byte
+
+	if s := ef.Section(".gopclntab"); s != nil {
+		if data, err = s.Data(maxBytesGoPclntab); err != nil {
+			return nil, fmt.Errorf("failed to load .gopclntab: %w", err)
+		}
+	} else if s := ef.Section(".data.rel.ro.gopclntab"); s != nil {
+		if data, err = s.Data(maxBytesGoPclntab); err != nil {
+			return nil, fmt.Errorf("failed to load .data.rel.ro.gopclntab: %w", err)
+		}
+	} else if s := ef.Section(".go.buildinfo"); s != nil {
+		symtab, err := ef.ReadSymbols()
+		if err != nil {
+			// It seems the Go binary was stripped. So we use the heuristic approach
+			// to get the stack deltas.
+			if data, err = elfunwindinfo.SearchGoPclntab(ef); err != nil {
+				return nil, fmt.Errorf("failed to search .gopclntab: %w", err)
+			}
+		} else {
+			start, err := symtab.LookupSymbolAddress("runtime.pclntab")
+			if err != nil {
+				return nil, fmt.Errorf("failed to load .gopclntab via symbols: %w", err)
+			}
+			end, err := symtab.LookupSymbolAddress("runtime.epclntab")
+			if err != nil {
+				return nil, fmt.Errorf("failed to load .gopclntab via symbols: %w", err)
+			}
+			if start >= end {
+				return nil, fmt.Errorf("invalid .gopclntab symbols: %v-%v", start, end)
+			}
+			data = make([]byte, end-start)
+			if _, err := ef.ReadVirtualMemory(data, int64(start)); err != nil {
+				return nil, fmt.Errorf("failed to load .gopclntab via symbols: %w", err)
+			}
+		}
+	}
+
+	if data == nil {
+		return nil, nil
+	}
```

Comment:
The semantics of this function are:
* if the file does not appear to be a Go binary (no .gopclntab nor .go.buildinfo section), return a nil data without error 
* if the file appear to be a Go binary but .gopclntab extraction fails for another reason, return an error
* otherwise return found .gopclntab

The idea being that it could be interesting to log the error cases.